src/Compilers/CSharp/Portable/Symbols/Source/SourcePropertyAccessorSymbol.cs

```diff
@@ -213,7 +209,8 @@ internal sealed override ImmutableArray<string> NotNullWhenFalseMembers
                    isIterator)
         {
             _property = property;
-            _isAutoPropertyAccessor = isAutoPropertyAccessor;
+            _accessorBodyShouldBeSynthesized = (property.IsAutoProperty && syntax is AccessorDeclarationSyntax { SemicolonToken.RawKind: (int)SyntaxKind.SemicolonToken, ExpressionBody: null, Body: null }) ||
```

Comment:
>SemicolonToken.RawKind: (int)SyntaxKind.SemicolonToken [](http://example.com/codeflow?start=113&length=54)

It doesn't look like we actually need to check anything for the semicolon token itself. ```SourcePropertySymbol.GetAccessorDeclarations``` only checks if we have any explicit body. Let's be consistent and avoid any possible deviations around error scenarios.